server/src/main/java/org/elasticsearch/index/MergePolicyConfig.java

```diff
@@ -120,6 +123,45 @@ public final class MergePolicyConfig {
         Property.IndexScope
     );
 
+    public enum Type {
+        UNSET {
+            @Override
+            MergePolicy getMergePolicy(MergePolicyConfig config, boolean isTimeSeriesIndex) {
+                if (isTimeSeriesIndex) {
+                    // TieredMergePolicy is better than LogByteSizeMergePolicy at computing cheaper merges, but it does so by allowing
```

Comment:
This is formulated as if it in practice could be (substantially) more expensive to use the log merge policy, perhaps we can elaborate here on why this is unlikely to be the case for data streams?

Reply:
Agreed, I pushed an update that makes it sound like a better trade-off.